src/Psy/VersionUpdater/Checker.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/*
+ * This file is part of Psy Shell.
+ *
+ * (c) 2012-2015 Justin Hileman
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Psy\VersionUpdater;
+
+use GuzzleHttp\ClientInterface;
+use Psy\Shell;
+use Psy\VersionUpdater\Entity\Version;
+
+class Checker
+{
+    const METHOD = 'GET';
+    const URL = 'https://api.github.com/repos/bobthecow/psysh/releases/latest';
+
+    /** @var ClientInterface */
+    private $client;
+    /** @var Version */
+    private $current;
+    /** @var Version */
+    private $latest;
+
+    public function __construct(ClientInterface $client)
+    {
+        $this->client = $client;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isLatest()
+    {
+        try {
+            $this->current = new Version(Shell::VERSION);
+            $this->latest = new Version($this->getTagName());
+
+            return $this->current->compare($this->latest);
+        } catch (\InvalidArgumentException $e) {
+            return true;
```

Comment:
if it fails, "unable to check for updates" is probably a better message than "yep! totally up to date!"